platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/config/FileSystemManagerConfig.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2016-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.io.config;
+
+import static com.swirlds.common.io.utility.FileUtils.getAbsolutePath;
+
+import com.swirlds.common.config.StateCommonConfig;
+import com.swirlds.common.io.filesystem.FileSystemManager;
+import com.swirlds.common.platform.NodeId;
+import com.swirlds.config.api.ConfigData;
+import com.swirlds.config.api.ConfigProperty;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.nio.file.Path;
+import java.time.Duration;
+
+/**
+ * Settings for {@link FileSystemManager}
+ *
+ * @param rootPath
+ * 		The directory where temporary files are created.
+ * @param recycleBinMaximumFileAge   the maximum age of a file in the recycle bin before it is deleted
+ * @param recycleBinCollectionPeriod the period between recycle bin collection runs
+ */
+@ConfigData("fileSystemManager")
+public record FileSystemManagerConfig(
+        @ConfigProperty(defaultValue = "swirlds-root") String rootPath,
+        @ConfigProperty(defaultValue = "7d") Duration recycleBinMaximumFileAge,
+        @ConfigProperty(defaultValue = "1d") Duration recycleBinCollectionPeriod) {
+
+    /**
+     * Returns the real path for the root that depends on the {@link StateCommonConfig#savedStateDirectory()}
+     * property.
+     *
+     * @param stateConfig
+     * 		the state config object
+     * @return the location where temporary files are stored
+     */
+    public Path getRootPath(final StateCommonConfig stateConfig) {
+        return getAbsolutePath(stateConfig.savedStateDirectory().resolve(rootPath()));
+    }
+
+    /**
+     * Returns the real path for the root of the filesystem that depends both on the {@link StateCommonConfig#savedStateDirectory()}
+     * property and the {@link NodeId}
+     *
+     *
+     * @param stateConfig the state config object
+     * @param selfId      the ID of this node
+     * @return the location where recycle bin files are stored
+     */
+    public Path getRootPath(@NonNull final StateCommonConfig stateConfig, @NonNull final NodeId selfId) {
+        return getAbsolutePath(stateConfig
+                .savedStateDirectory()
+                .resolve(Long.toString(
+                        selfId.id())) // Or is the node the first parameter ?? how do we write the states files?
+                .resolve(rootPath()));
+    }
```

Comment:
Currently, the files are inside the  .savedStateDirectory(), which matches how the folder structure works today. We discussed with Kelly not to break the current folder structure for the node, but if I get you correctly, then the desired structure would be:

```
* node-root (it might contain the node id)
   * savedStateDirectory
   * tmp
   * usr
   * recycle-bin
```
FS-manager root matches node-root   
   
If I do that then It impacts how we configure the node, so now is
```
* savedStateDirectory
   *fs-manager-root
      * tmp
      * usr
      * recycle-bin  
```
Do you think is ok for now

Reply:
In the short term, we absolutely must not change the directories that outside processes are utilizing. A few examples that come to mind:

- saved states
- event stream
- record stream

Directories that are managed 100% by the java code are ok to change, as log as there is sufficient testing, and if we do appropriate migration and cleanup. E.g.

- temporary files
- recycle bin
- preconsensus event stream
- address book debug files

I suggest an incremental adoption of the new file system manager. When we first deploy this, we should accept the fact that not all files will utilize this tool immediately. Then, we should target the legacy files one at a time and come up with migration solutions.

One nice to have, if we could use configuration to change the "usr" directory name to a different name, we could have the same file path for a lot of important files. Make "data" the root directory, and "saved" the usr directory. That way, anything created by `FileSystemManager.resolve()` will live beneath `data/saved` (i.e. where most stuff currently lives).

Note that we will need special cleanup logic for the old temporary file directory and recycle bin directory during the first release where this is deployed. Normally cleanup is done at system boot time. But since we are changing those paths, the regular workflow won't know where to clean up.

Reply:
Ok! I like this, and it is a similar approach to what has been discussed internally with the Base Team

Reply:
Would the things inside "saved" need to be deleted at startup? or would that only stand for the temp dir?

Reply:
The root is now data/ and the user-data dir matches saved